lib/navigation/preprocess_encoders.cpp

```diff
@@ -1,91 +1,94 @@
+#include "consts.hpp"
 #include "preprocess_encoders.hpp"
 
+#include "core/types.hpp"
+
 namespace hyped::navigation {
 
 EncodersPreprocessor::EncodersPreprocessor(core::ILogger &logger)
     : logger_(logger),
-      encoder_outliers_({0, 0, 0, 0}),
-      encoders_reliable_({true, true, true, true}),
-      num_reliable_encoders_(core::kNumImus)
+      num_consecutive_outliers_per_encoder_({0, 0, 0, 0}),
+      is_reliable_per_encoder_({true, true, true, true}),
+      num_reliable_encoders_(core::kNumImus),
+      max_consecutive_outliers_(10)
 {
 }
 
 std::optional<core::EncoderData> EncodersPreprocessor::processData(
-  const core::EncoderData raw_encoder_data)
+  const core::EncoderData &raw_encoder_data)
 {
-  const auto encoder_data = detectOutliers(raw_encoder_data);
-  SensorChecks sensors    = checkReliable();
-  if (!encoder_data.has_value() || sensors == SensorChecks::kUnacceptable) {
-    return std::nullopt;
-  } else {
-    return encoder_data;
-  }
+  const auto encoder_data = sanitise(raw_encoder_data);
+  const auto sensors      = checkReliable();
+  if (sensors == SensorChecks::kUnacceptable) { return std::nullopt; }
+  return encoder_data;
 }
 
-std::optional<core::EncoderData> EncodersPreprocessor::detectOutliers(
-  const core::EncoderData encoder_data)
+std::optional<EncodersPreprocessor::Statistics> EncodersPreprocessor::getStatistics(
```

Comment:
Not a fan of "statistics" it doesn't really describe functionality (we specifically calculate quartiles)?

Reply:
the input to this function is a set of data, the output is a median, and thresholds

this has nothing to do with quartiles from the callers perspective

i agree, though; there's a TODO comment indicating that I don't like the name

we could go for `MedianAndThresholds` if there isn't a better option